src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs

```diff
@@ -31,6 +35,20 @@ public ConsoleLogRecordExporter(ConsoleExporterOptions options)
 
         public override ExportResult Export(in Batch<LogRecord> batch)
         {
+            if (this.disposed)
+            {
+                if (!this.isDisposeMessageSent)
+                {
+                    this.isDisposeMessageSent = true;
+                    this.WriteLine("The console exporter is still being invoked after it has been disposed. This could be the result of invalid lifecycle management of the OpenTelemetry .NET SDK.");
+                    this.WriteLine(Environment.StackTrace);
+                    this.WriteLine(Environment.NewLine + "Dispose was called on the following stack trace:");
+                    this.WriteLine(this.disposedStackTrace);
+                }
+
+                return ExportResult.Failure;
+            }
```

Comment:
This code looks buggy to me.

I think the correct logic should be something like:
```csharp
if (disposed && !isDisposeMessageSent)
{
    lock (syncObject)
    {
        if (isDisposeMessageSent)
        {
            return Failure;
        }
    }

    WriteLine(...);
}

Reply:
sorry i misunderstood what you meant, update the code, thank you